src/transformers/cache_utils.py

```diff
@@ -777,7 +786,20 @@ def __init__(self, cache_config: CacheConfig) -> None:
 
     def _quantize(self, tensor, axis):
         scale, zeropoint = self.optimizer(tensor, self.qtype.bits, axis, self.q_group_size)
-        qtensor = AffineQuantizer.apply(tensor, self.qtype, axis, self.q_group_size, scale, zeropoint)
+        # We have two different API since in optimum-optimun, we don't use AffineQuantizer anymore
+        if is_optimum_quanto_available():
+            from optimum.quanto import QBitsTensor
+
+            qtensor = QBitsTensor.quantize(tensor, self.qtype, axis, self.q_group_size, scale, zeropoint)
```

Comment:
Here you should call the public API which is `quantize_weight` or `WeightQBitsTensor.quantize` but this is more an internal API.